python/ray/tune/util.py

```diff
@@ -18,6 +20,59 @@
 PINNED_OBJECT_PREFIX = "ray.tune.PinnedObject:"
 
 
+class UtilMonitor(Thread):
+    """Class for system usage utilization monitoring.
+
+    It keeps track of CPU, RAM, GPU, VRAM usage (each gpu separately) by
+    pinging for information every x seconds in a separate thread.
+    """
+    def __init__(self, delay=0.7):
+
+        try:
+            global psutil
+            import psutil
+            global GPUtil
+            import GPUtil
+        except ModuleNotFoundError:
+            logging.warning("System performance monitoring will not be available."
+                            "Install psutil and gputil packages.")
+            self.stopped = True
+            return
+
+        super(UtilMonitor, self).__init__()
+        self.stopped = False
+        self.delay = delay  # Time between calls to GPUtil
+        self.values = defaultdict(list)
+        self.lock = threading.Lock()
+        self.daemon = True
+        self.start()
+
+    def read_utilization(self):
+        with self.lock:
+            self.values["perf/cpu"].append(float(psutil.cpu_percent(interval=None)))
+            self.values["perf/ram"].append(float(getattr(psutil.virtual_memory(), 'percent')))
+            for gpu in GPUtil.getGPUs():
+                self.values["perf/gpu" + str(gpu.id)].append(float(gpu.load))
+                self.values["perf/vram" + str(gpu.id)].append(float(gpu.memoryUtil))
+
+    def get_data(self):
+        if self.stopped:
+            return {}
+
+        with self.lock:
+            ret_values = copy.deepcopy(self.values)
+            for key, val in self.values.items():
+                val.clear()
+        return {k: np.mean(v) for k, v in ret_values.items()}
```

Comment:
I noticed I sometimes get 
```
  perf/cpu: .nan
  perf/ram: .nan
```

Is there a way to avoid this? Maybe report the same result as the last interval if empty?